pkgs/tools/virtualization/nixos-container/nixos-container.pl

```diff
@@ -309,14 +290,14 @@ sub safeRemoveTree {
 }
 
 elsif ($action eq "root-login") {
-    runInContainer("@su@", "root", "-l");
+    exec("machinectl", "shell", $containerName, "/bin/sh", "-l");
 }
 
 elsif ($action eq "run") {
     shift @ARGV; shift @ARGV;
     # Escape command.
     my $s = join(' ', map { s/'/'\\''/g; "'$_'" } @ARGV);
-    runInContainer("@su@", "root", "-l", "-c", "exec " . $s);
+    exec("machinectl", "--quiet", "shell", $containerName, "/bin/sh", "-l", "-c", $s);
```

Comment:
It would be cool, if the shell would not interpret the arguments
passed to the commands. 
Otherwise people have to use double escaping.
I only fear this would break backward compatibility:

Ideally the shell would receive the arguments in the following way.

```perl
exec("machinectl", "--quiet", "shell", $containerName, "/bin/sh", "-l", "-c", 'exec "$@"', @ARGV);
```

I am not sure, if this is the correct perl syntax to achieve, what I want.